_posts/2018-09-13-is-cockpit-secure.md

```diff
@@ -0,0 +1,257 @@
+you make those choices. You'll find not only has Cockpit got a solid security
+story, but you can use it in all sorts of different ways depending on what
+kind of security your systems need.
+
+## Cockpit is a Linux session in your browser
+
+Let start by stacking the deck against Cockpit. First off it's important to
+remember that Cockpit is actually an alternative Linux session ... along side X11,
+SSH and VT logins. It's a session in the [PAM](http://www.linux-pam.org/),
+[logind](https://www.freedesktop.org/wiki/Software/systemd/logind/), TTY,
+[SELinux](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Security-Enhanced_Linux/sect-Security-Enhanced_Linux-Confining_Users-Confining_Existing_Linux_Users_semanage_login.html)
+ and other senses.
+The logged in browser and the javascript code it is running can interact directly
+with the system as part of that session.
+
+Try this out if you want. [Log into Cockpit](http://cockpit-project.org/running.html),
+and pop up the Javascript console by pressing *Ctrl-Shift-J* in your browser. Now type
+commands like the following:
+
+    > zz = cockpit.spawn(["id"])
+    > proxy = cockpit.dbus("org.freedesktop.hostname1").proxy()
+    > proxy.KernelName
+
+Pretty cool, huh. And that's why Cockpit can be such a powerful and quick way to build
+Linux system admin interfaces.
+
+## But are browsers secure?
+
+But can we trust browsers with your security? Firefox is about
+[15 million lines of code](https://www.openhub.net/p/firefox/analyses/latest/languages_summary)
+that all have to work together, and after looking at the list of
+[security updates](https://www.mozilla.org/en-US/security/known-vulnerabilities/firefox/)
+you gotta think about what that means for your security.
+
+Any management system with a browser interface requires a secure browser.
+That includes
+[Satellite](https://www.redhat.com/en/technologies/management/satellite),
+[Tower](https://www.ansible.com/tower),
+[Foreman](https://www.theforeman.org/),
+[Landscape](https://landscape.canonical.com/),
+[CloudForms](https://www.redhat.com/en/technologies/management/cloudforms),
+[cPanel](https://cpanel.com/) and more. If anything like this interacts with
+your systems, then security bugs in browsers have a direct effect on your
+system security.
+
+Secondly take a look at any browser based tools that were involved in the
+creation or curation of the software installed on your system, including
+tools used by your Linux distribution.
+
+If your security requirements are strict and you must avoid browsers at all
+costs, then be sure to audit all possible places that browsers were involved
+in what's running on your systems. It's harder than you think.
+
+Unless your system is very isolated, it turns out that browsers are
+likely already highly involved in the security paths for your systems.
+Cockpit does not change this situation significantly.
+
```

Comment:
Does it make sense to mention the possibility of using a bastion host here?

Reply:
This security consideration has nothing to do with bastion hosts. Your browser is your browser. But we do mention bastion hosts below.